options/options.go

```diff
@@ -0,0 +1,23 @@
+package options
+
+type validationFunc func(val interface{}) bool
+
+type optionDetails struct {
+	defaultValue    interface{}
+	validationFuncs []validationFunc
+	help            string
+}
+
```

Comment:
@kshlm wanted to avoid having duplication of xlator options. I can't think of any other way though, especially the options that changes the volfile graph seems needed here.